testcase.py

```diff
@@ -1054,6 +1062,100 @@ def get_version(self):
 
         return status, message, payload
 
+    def post_results_to_dashboard(self, api_key, tags):
+        '''Posts test results to online dashboard at given server address.
+
+        Parameters
+        ----------
+        api_key : str
+            API key corresponding to user account for dashboard.
+        tags : list of str
+            List of tags to be included with result posting.
+
+        Returns
+        -------
+        status: int
+            Indicates whether a posting to the dashboard has been successfully completed.
+            If 200, the posting was successful.
+            If 400, there was an input error.
+            If 500, an internal error occurred.
+        message: str
+            Includes detailed debugging information
+        payload: None
+
+        '''
+
+        # Check formal scnenario test completed
+        if not self.scenario_end:
```

Comment:
I expect Devetry is doing some validation on the payload and returning error codes, but I see no reason not to do this pre checks.

Reply:
I'm not sure there's anything stopping someone from starting a scenario, running it halfway, and then submitting results, in which case all the payload data would be "valid", but the KPIs would be non-comparable.  This is to make sure a scenario is finished (i.e. time period) before allowing to send results.  There is some error checking on the dashboard side and I handle that starting at https://github.com/ibpsa/project1-boptest/blob/issue403_submit/testcase.py#L1146.